packages/web-components/fast-foundation/src/anchored-region/anchored-region.template.ts

```diff
@@ -2,7 +2,10 @@ import { html, ref, when } from "@microsoft/fast-element";
 import { AnchoredRegion } from "./anchored-region";
 
 export const AnchoredRegionTemplate = html<AnchoredRegion>`
-    <div class="region" part="region" ${ref("region")} style=${x => x.regionStyle}>
-        ${when(x => x.initialLayoutComplete, html`<slot></slot>`)}
-    </div>
+    <template ${ref("root")}>
+        <div class="region" part="region" ${ref("region")} style=${x => x.regionStyle}>
+            ${when(x => x.initialLayoutComplete, html<AnchoredRegion>`<slot></slot>`)}
+        </div>
+        <template> </template
```

Comment:
If we made calls to getBoundingClientRect would we need all this markup? I'm concerned that in our attempt to avoid calls to gBCR we're forcing people to work with DOM they would otherwise not include. It seems like that might solve some of the issues we are running up against.

Full transparency, I would take calls to gBCR versus the prescriptive DOM we have here. I'm curious how much it's creating the complexity that's being added below. While I understand that this does complex things, I think we're exacerbating it by doing everything in our power to avoid gBCR.

Reply:
I don't think using gBCR would change the reasoning behind the markup, it could perhaps simplify some of the component's internal logic but I"m not sure we could avoid calling it too much.

Reply:
I may not understand the reasoning for the markup, perhaps you can elaborate on it? 

I think gBCR would end up providing support for Fixed position items where we currently don't have that. This would end up being helpful in scenarios where you have a fixed or sticky header and need to flyout an item from it.

Reply:
Primarily, the template wrapper around the region provides a known coordinate space to move the region around in, if we did not have this we'd have to suss out the parent's coordinate space and the region's position within it.